src/checks/taskChecks.ts

```diff
@@ -0,0 +1,26 @@
+import {AslChecker, StateMachineErrorCode} from "../types";
+import {checkPropertyWorkflow} from "./checkProperty";
+
+export const taskChecks: AslChecker = (definition) => {
+    return [
+        ...checkPropertyWorkflow({
+            definition,
+            filter: ({state}) => {
+                return state.Type === 'Task'
+            },
+            propCheck: 'atMostOne',
+            props: ["TimeoutSeconds", "TimeoutSecondsPath"],
+            errorCode: StateMachineErrorCode.TaskTimeoutError,
+        }),
+
+        ...checkPropertyWorkflow({
+            definition,
+            filter: ({state}) => {
+                return state.Type === 'Task'
+            },
+            propCheck: 'atMostOne',
+            props: ["HeartbeatSeconds", "HeartbeatSecondsPath"],
+            errorCode: StateMachineErrorCode.TaskHeartbeatError,
+        })
```

Comment:
two new checks for the task's timeout and heartbeat properties 

these weren't being checked before so no changes needed in the schema